lib/shopify-cli/shopifolk.rb

```diff
@@ -0,0 +1,70 @@
+module ShopifyCli
+  ##
+  # ShopifyCli::Shopifolk contains the logic to determine if the user appears to be a Shopify staff
+  #
+  # The Shopifolk Feature flag will persist between runs so if the flag is enabled or disabled,
+  # it will still be in that same state until the next class invocation.
+  class Shopifolk
+    GCLOUD_CONFIG_PATH = '~/.config/gcloud/configurations/config_default'
+    DEV_PATH = '/opt/dev'
+    SECTION = 'core'
+    FEATURE_NAME = 'shopifolk'
+    def self.check
+      ##
+      # will return if the user appears to be a Shopify employee, based on several heuristics
+      #
+      # #### Returns
+      #
+      # * `is_shopifolk` - returns true if the user is a Shopify Employee
+      #
+      # #### Example
+      #
+      #     ShopifyCli::Shopifolk.check
+      #
+      ShopifyCli::Shopifolk.new.shopifolk?
+    end
+
+    def shopifolk?(gcloud_config_path = GCLOUD_CONFIG_PATH, dev_path = DEV_PATH, debug = false)
+      ##
+      # will return if the user is a Shopify employee
+      #
+      # #### Returns
+      #
+      # * `is_shopifolk` - returns true if the user has `dev` installed and
+      # a valid google cloud config file with email ending in "@shopify.com"
+      #
+      unless debug
+        return true if Feature.enabled?(FEATURE_NAME)
```

Comment:
This is not so efficient for non-Shopify users, since it will be checking everytime but let's address that in a follow-up.

Reply:
Ooh I see what you mean. I'll keep that in mind